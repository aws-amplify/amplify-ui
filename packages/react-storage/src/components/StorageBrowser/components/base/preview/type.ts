export interface PreviewComponentProps {
  url: string;
  fileKey: string;
  showDownload?: boolean;
}

export type PreviewComponent = React.ComponentType<PreviewComponentProps>;
