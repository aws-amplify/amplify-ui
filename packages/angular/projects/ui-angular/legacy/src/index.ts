export {
  AmplifyButton,
  AmplifyChatbot,
  AmplifyIcon,
  AmplifyInput,
  AmplifyPhotoPicker,
  AmplifyPicker,
  AmplifyS3Album,
  AmplifyS3Image,
  AmplifyS3ImagePicker,
  AmplifyS3Text,
  AmplifyS3TextPicker,
  AmplifyToast,
} from './proxies';

export { LegacyAmplifyUiModule } from './legacy-ui-angular.module';
