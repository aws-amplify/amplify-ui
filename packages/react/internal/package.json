{
  "private": true,
  "name": "@aws-amplify/ui-react-internal",
  "main": "../dist/internal.js",
  "module": "../dist/esm/internal.mjs",
  "types": "../dist/types/internal.d.ts"
}
