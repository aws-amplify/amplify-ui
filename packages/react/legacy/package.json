{
  "private": true,
  "name": "@aws-amplify/ui-react-legacy",
  "main": "../dist/legacy.js",
  "module": "../dist/esm/legacy.js",
  "types": "../dist/types/legacy.d.ts"
}
