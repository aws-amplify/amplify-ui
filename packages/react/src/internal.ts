export * from './hooks/useAuth';
export * from './hooks/useDataStore';
export * from './hooks/useStorageURL';
export * from './primitives/Icon/internal';
export {
  UseAuthSignOutAction,
  useAuthSignOutAction,
} from './hooks/actions/useAuthSignOutAction';
export {
  useNavigateAction,
  UseNavigateActionOptions,
} from './hooks/actions/useNavigateAction';
export { useStateMutationAction } from './hooks/actions/useStateMutationAction';

export { useDataStoreCreateAction } from './hooks/actions/useDataStoreCreateAction';
export type { UseDataStoreCreateActionOptions } from './hooks/actions/useDataStoreCreateAction';
export { useDataStoreDeleteAction } from './hooks/actions/useDataStoreDeleteAction';
export type { UseDataStoreDeleteActionOptions } from './hooks/actions/useDataStoreDeleteAction';
export { useDataStoreUpdateAction } from './hooks/actions/useDataStoreUpdateAction';
export type { UseDataStoreUpdateActionOptions } from './hooks/actions/useDataStoreUpdateAction';

export * from './primitives/shared/datastore';
export {
  EscapeHatchProps,
  Variant,
  findChildOverrides,
  getOverridesFromVariants,
  getOverrideProps,
  mergeVariantsAndOverrides,
} from './studio';

export { PrimitiveCatalog } from './PrimitiveCatalog';
