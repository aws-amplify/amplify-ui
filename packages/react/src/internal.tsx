export * from './hooks/useAuth';
export * from './hooks/useDataStore';
export * from './hooks/useStorageURL';
export * from './primitives/shared/datastore';
export {
  EscapeHatchProps,
  Variant,
  findChildOverrides,
  getOverridesFromVariants,
  getOverrideProps,
  mergeVariantsAndOverrides,
} from './primitives/shared/utils';
