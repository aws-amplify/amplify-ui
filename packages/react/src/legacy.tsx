export {
  /** @deprecated `AmplifyAuthenticator` is the v1 release. Prefer the latest `Authenticator` for continued support. */
  AmplifyAuthenticator,
  AmplifySignIn,
  AmplifySignOut,
  AmplifyChatbot,
  AmplifyPhotoPicker,
  AmplifyPicker,
  AmplifyS3Album,
  AmplifyS3Image,
  AmplifyS3ImagePicker,
  AmplifyS3Text,
  AmplifyS3TextPicker,
  /** @deprecated `withAuthenticator` from legacy. Prefer the latest `withAuthenticator` release for continued support. */
  withAuthenticator,
} from '@aws-amplify/ui-react-v1';
