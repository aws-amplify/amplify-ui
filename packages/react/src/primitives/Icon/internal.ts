export * from './icons';
export { useIcons } from './context/useIcons';
