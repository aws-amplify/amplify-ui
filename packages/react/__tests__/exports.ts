import fs from 'fs';
import path from 'path';
import type { PrimitiveCatalogType } from '../src/types/catalog';
import { PrimitiveCatalog } from '@aws-amplify/ui-react/internal';

// Jest doesn't support `exports` maps, so we have to reference `dist` directly.
// See: https://github.com/facebook/jest/issues/9771
import * as exported from '../dist';
import * as legacy from '../dist/legacy';
import * as internal from '../dist/internal';

describe('@aws-amplify/ui-react', () => {
  describe('exports', () => {
    it('should match snapshot', () => {
      expect(Object.keys(exported)).toMatchInlineSnapshot(`
        Array [
          "createTheme",
          "defaultDarkModeOverride",
          "defaultTheme",
          "translations",
          "ComponentClassNames",
          "ComponentClassObject",
          "ComponentPropsToStylePropsMap",
          "ComponentPropsToStylePropsMapKeys",
          "View",
          "useTheme",
          "Alert",
          "AmplifyProvider",
          "Authenticator",
          "Badge",
          "Button",
          "ButtonGroup",
          "Card",
          "CheckboxField",
          "Collection",
          "Divider",
          "Expander",
          "ExpanderItem",
          "FieldGroupIcon",
          "FieldGroupIconButton",
          "Flex",
          "Geocoder",
          "Grid",
          "Heading",
          "Icon",
          "Image",
          "Link",
          "Loader",
          "LocationSearch",
          "MapView",
          "Menu",
          "MenuButton",
          "MenuItem",
          "Pagination",
          "PasswordField",
          "PhoneNumberField",
          "Placeholder",
          "Radio",
          "RadioGroupField",
          "Rating",
          "ScrollView",
          "SearchField",
          "SelectField",
          "SliderField",
          "StepperField",
          "SwitchField",
          "TabItem",
          "Table",
          "TableBody",
          "TableCell",
          "TableFoot",
          "TableHead",
          "TableRow",
          "Tabs",
          "Text",
          "TextAreaField",
          "TextField",
          "ThemeProvider",
          "ToggleButton",
          "ToggleButtonGroup",
          "VisuallyHidden",
          "components",
          "primitives",
          "useAmplify",
          "useAuthenticator",
          "useBreakpointValue",
          "usePagination",
          "withAuthenticator",
        ]
      `);
    });
  });
});

describe('@aws-amplify/ui-react/legacy', () => {
  describe('exports', () => {
    it('should match snapshot', () => {
      expect(Object.keys(legacy)).toMatchInlineSnapshot(`
        Array [
          "AmplifyAuthenticator",
          "AmplifyChatbot",
          "AmplifyPhotoPicker",
          "AmplifyPicker",
          "AmplifyS3Album",
          "AmplifyS3Image",
          "AmplifyS3ImagePicker",
          "AmplifyS3Text",
          "AmplifyS3TextPicker",
          "AmplifySignIn",
          "AmplifySignOut",
          "withAuthenticator",
        ]
      `);
    });
  });
});

describe('@aws-amplify/ui-react/internal', () => {
  describe('exports', () => {
    it('should match snapshot', () => {
      expect(Object.keys(internal)).toMatchInlineSnapshot(`
        Array [
          "IconAdd",
          "IconCheck",
          "IconCheckCircle",
          "IconChevronLeft",
          "IconChevronRight",
          "IconClose",
          "IconError",
          "IconExpandMore",
          "IconIndeterminate",
          "IconInfo",
          "IconMenu",
          "IconRemove",
          "IconSearch",
          "IconStar",
          "IconVisibility",
          "IconVisibilityOff",
          "IconWarning",
          "IconCheckCircleOutline",
          "IconFiberManualRecord",
          "IconHighlightOff",
          "PrimitiveCatalog",
          "createDataStorePredicate",
          "findChildOverrides",
          "getOverrideProps",
          "getOverridesFromVariants",
          "mergeVariantsAndOverrides",
          "useAuth",
          "useAuthSignOutAction",
          "useDataStoreBinding",
          "useDataStoreCollection",
          "useDataStoreCreateAction",
          "useDataStoreDeleteAction",
          "useDataStoreItem",
          "useDataStoreUpdateAction",
          "useNavigateAction",
          "useStateMutationAction",
          "useStorageURL",
        ]
      `);
    });
  });
});

describe('primitive catalog', () => {
  it.each(Object.entries(PrimitiveCatalog))(
    'should contain properties for %s primitive',
    (name, primitive) => {
      expect(Object.keys(primitive.properties).length).toBeGreaterThan(0);
    }
  );

  it('should match primitives list snapshot', () => {
    expect(Object.keys(PrimitiveCatalog)).toMatchInlineSnapshot(`
      Array [
        "Alert",
        "Badge",
        "Button",
        "ButtonGroup",
        "Card",
        "CheckboxField",
        "Collection",
        "Divider",
        "Expander",
        "ExpanderItem",
        "FieldGroupIcon",
        "FieldGroupIconButton",
        "Flex",
        "Grid",
        "Heading",
        "Icon",
        "Image",
        "Link",
        "Loader",
        "Menu",
        "MenuButton",
        "MenuItem",
        "Pagination",
        "PasswordField",
        "PhoneNumberField",
        "Placeholder",
        "Radio",
        "RadioGroupField",
        "Rating",
        "ScrollView",
        "SearchField",
        "SelectField",
        "SliderField",
        "StepperField",
        "SwitchField",
        "Tabs",
        "TabItem",
        "Text",
        "TextAreaField",
        "TextField",
        "ToggleButton",
        "ToggleButtonGroup",
        "View",
        "VisuallyHidden",
        "Table",
        "TableBody",
        "TableCell",
        "TableFoot",
        "TableHead",
        "TableRow",
      ]
    `);
  });
});
