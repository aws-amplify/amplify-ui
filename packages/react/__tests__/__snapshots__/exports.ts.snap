// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@aws-amplify/ui-react exports should match snapshot 1`] = `
Array [
  "Alert",
  "AmplifyProvider",
  "Authenticator",
  "Badge",
  "Button",
  "ButtonGroup",
  "Card",
  "CheckboxField",
  "Collection",
  "ComponentClassNames",
  "ComponentClassObject",
  "ComponentPropsToStylePropsMap",
  "ComponentPropsToStylePropsMapKeys",
  "Divider",
  "Expander",
  "ExpanderItem",
  "FieldGroupIcon",
  "FieldGroupIconButton",
  "Flex",
  "Geocoder",
  "Grid",
  "Heading",
  "Icon",
  "Image",
  "InAppMessageDisplay",
  "InAppMessagingProvider",
  "Link",
  "Loader",
  "LocationSearch",
  "MapView",
  "Menu",
  "MenuButton",
  "MenuItem",
  "Pagination",
  "PasswordField",
  "PhoneNumberField",
  "Placeholder",
  "Radio",
  "RadioGroupField",
  "Rating",
  "ScrollView",
  "SearchField",
  "SelectField",
  "SliderField",
  "StepperField",
  "SwitchField",
  "TabItem",
  "Table",
  "TableBody",
  "TableCell",
  "TableFoot",
  "TableHead",
  "TableRow",
  "Tabs",
  "Text",
  "TextAreaField",
  "TextField",
  "ThemeProvider",
  "ToggleButton",
  "ToggleButtonGroup",
  "View",
  "VisuallyHidden",
  "components",
  "createTheme",
  "defaultDarkModeOverride",
  "defaultTheme",
  "primitives",
  "translations",
  "useAmplify",
  "useAuthenticator",
  "useBreakpointValue",
  "useInAppMessaging",
  "usePagination",
  "useTheme",
  "withAuthenticator",
  "withInAppMessaging",
]
`;

exports[`@aws-amplify/ui-react/internal exports should match snapshot 1`] = `
Array [
  "IconAdd",
  "IconCheck",
  "IconCheckCircle",
  "IconCheckCircleOutline",
  "IconChevronLeft",
  "IconChevronRight",
  "IconClose",
  "IconError",
  "IconExpandMore",
  "IconFiberManualRecord",
  "IconHighlightOff",
  "IconIndeterminate",
  "IconInfo",
  "IconMenu",
  "IconRemove",
  "IconSearch",
  "IconStar",
  "IconVisibility",
  "IconVisibilityOff",
  "IconWarning",
  "PrimitiveCatalog",
  "createDataStorePredicate",
  "findChildOverrides",
  "getOverrideProps",
  "getOverridesFromVariants",
  "mergeVariantsAndOverrides",
  "useAuth",
  "useAuthSignOutAction",
  "useDataStoreBinding",
  "useDataStoreCollection",
  "useDataStoreCreateAction",
  "useDataStoreDeleteAction",
  "useDataStoreItem",
  "useDataStoreUpdateAction",
  "useNavigateAction",
  "useStateMutationAction",
  "useStorageURL",
]
`;

exports[`primitive catalog should match primitives list snapshot 1`] = `
Array [
  "Alert",
  "Badge",
  "Button",
  "ButtonGroup",
  "Card",
  "CheckboxField",
  "Collection",
  "Divider",
  "Expander",
  "ExpanderItem",
  "FieldGroupIcon",
  "FieldGroupIconButton",
  "Flex",
  "Grid",
  "Heading",
  "Icon",
  "Image",
  "Link",
  "Loader",
  "Menu",
  "MenuButton",
  "MenuItem",
  "Pagination",
  "PasswordField",
  "PhoneNumberField",
  "Placeholder",
  "Radio",
  "RadioGroupField",
  "Rating",
  "ScrollView",
  "SearchField",
  "SelectField",
  "SliderField",
  "StepperField",
  "SwitchField",
  "TabItem",
  "Table",
  "TableBody",
  "TableCell",
  "TableFoot",
  "TableHead",
  "TableRow",
  "Tabs",
  "Text",
  "TextAreaField",
  "TextField",
  "ToggleButton",
  "ToggleButtonGroup",
  "View",
  "VisuallyHidden",
]
`;
